src/ui_manager.py

```diff
@@ -1,442 +1,456 @@
+        keyboard.send("esc")
+        # just in case also bring cursor to center and click
+        x, y = self._screen.convert_screen_to_monitor((self._config.ui_pos["center_x"], self._config.ui_pos["center_y"]))
+        mouse.move(x, y, randomize=25, delay_factor=[1.0, 1.5])
+        mouse.click(button="left")
+
+    def repair_and_fill_up_tp(self) -> bool:
+        """
+        Repair and fills up TP buy selling tome and buying. Vendor inventory needs to be open!
+        :return: Bool if success
+        """
+        repair_btn = self._template_finder.search_and_wait("REPAIR_BTN", roi=self._config.ui_roi["repair_btn"], time_out=4)
+        if not repair_btn.valid:
+            return False
+        x, y = self._screen.convert_screen_to_monitor(repair_btn.position)
+        mouse.move(x, y, randomize=12, delay_factor=[1.0, 1.5])
+        wait(0.1, 0.15)
+        mouse.click(button="left")
+        wait(0.1, 0.15)
+        x, y = self._screen.convert_screen_to_monitor((self._config.ui_pos["vendor_misc_x"], self._config.ui_pos["vendor_misc_y"]))
+        mouse.move(x, y, randomize=[20, 6], delay_factor=[1.0, 1.5])
+        wait(0.1, 0.15)
+        mouse.click(button="left")
+        wait(0.5, 0.6)
+        # Section modified by Cutebeast to check when Tome is empty
+        tp_tome = self._template_finder.search_and_wait(["TP_TOME", "TP_TOME_RED"], roi=self._config.ui_roi["inventory"], time_out=3)
+        if not tp_tome.valid:
+            return False
+        x, y = self._screen.convert_screen_to_monitor(tp_tome.position)
+        keyboard.send('ctrl', do_release=False)
+        mouse.move(x, y, randomize=8, delay_factor=[1.0, 1.5])
+        wait(0.1, 0.15)
+        mouse.press(button="left")
+        wait(0.25, 0.35)
+        mouse.release(button="left")
+        wait(0.5, 0.6)
+        keyboard.send('ctrl', do_press=False)
+        tp_tome = self._template_finder.search_and_wait("TP_TOME", roi=self._config.ui_roi["vendor_stash"], time_out=3)
+        if not tp_tome.valid:
+            return False
+        x, y = self._screen.convert_screen_to_monitor(tp_tome.position)
+        keyboard.send('ctrl', do_release=False)
+        mouse.move(x, y, randomize=8, delay_factor=[1.0, 1.5])
+        wait(0.1, 0.15)
+        mouse.click(button="right")
+        wait(0.1, 0.15)
+        keyboard.send('ctrl', do_press=False)
+        # delay to make sure the tome has time to transfer to other inventory before closing window
+        tp_tome = self._template_finder.search_and_wait("TP_TOME", roi=self._config.ui_roi["inventory"], time_out=3)
+        if not tp_tome.valid:
+            return False
+        return True
+
+    # Start Section by Cutebeast: Checking for Tome Condition
+    # Return True if no more TP
+    def no_more_tp(self) -> bool:
+        keyboard.send(self._config.char["tp"])
+        wait(0.02, 0.05)
+        tome_active = self._template_finder.search_and_wait("TOME_ACTIVE", roi=self._config.ui_roi["skill_right"], time_out=4)

```

Comment:
So we check if we have an inactive tomb, right? Since the template is red. That means we have an added delay of 4 seconds after each run?

Also having a "no" in the function and returning "False". Usually it is easier to read to have "has_tps()" and return True.

Reply:
ok i will fix it up tomorrow